RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc

```diff
@@ -72,13 +85,29 @@ BeamSpotOnlineProducer::BeamSpotOnlineProducer(const ParameterSet& iconf)
     : changeFrame_(iconf.getParameter<bool>("changeToCMSCoordinates")),
       theMaxZ(iconf.getParameter<double>("maxZ")),
       theSetSigmaZ(iconf.getParameter<double>("setSigmaZ")),
-      useTransientRecord_(iconf.getParameter<bool>("useTransientRecord")),
-      scalerToken_(useTransientRecord_ ? edm::EDGetTokenT<BeamSpotOnlineCollection>()
+      timeThreshold_(iconf.getParameter<int>("timeThreshold")),
+      sigmaZThreshold_(iconf.getParameter<double>("sigmaZThreshold")),
+      sigmaXYThreshold_(iconf.getParameter<double>("sigmaXYThreshold") * 1E-4),
+      useBSOnlineRecords_(iconf.getParameter<bool>("useBSOnlineRecords")),
+      scalerToken_(useBSOnlineRecords_ ? edm::EDGetTokenT<BeamSpotOnlineCollection>()
                                        : consumes<BeamSpotOnlineCollection>(iconf.getParameter<InputTag>("src"))),
       l1GtEvmReadoutRecordToken_(consumes<L1GlobalTriggerEvmReadoutRecord>(iconf.getParameter<InputTag>("gtEvmLabel"))),
       beamToken_(esConsumes<BeamSpotObjects, BeamSpotObjectsRcd>()),
-      beamTransientToken_(esConsumes<BeamSpotObjects, BeamSpotTransientObjectsRcd>()),
+      beamTokenLegacy_(esConsumes<BeamSpotOnlineObjects, BeamSpotOnlineLegacyObjectsRcd>()),
+      beamTokenHLT_(esConsumes<BeamSpotOnlineObjects, BeamSpotOnlineHLTObjectsRcd>()),
       theBeamShoutMode(iconf.getUntrackedParameter<unsigned int>("beamMode", 11)) {
+  fakeBS_.setBeamWidthX(0.1);
+  fakeBS_.setBeamWidthY(0.1);
+  fakeBS_.setSigmaZ(15.);
+  fakeBS_.setPosition(0.0001, 0.0001, 0.0001);
+  fakeBS_.setType(reco::BeamSpot::Fake);
+  fakeBS_.setCovariance(0, 0, 5e-10);
+  fakeBS_.setCovariance(1, 1, 5e-10);
+  fakeBS_.setCovariance(2, 2, 0.002);
+  fakeBS_.setCovariance(3, 3, 0.002);
+  fakeBS_.setCovariance(4, 4, 5e-11);
+  fakeBS_.setCovariance(5, 5, 5e-11);
+  fakeBS_.setCovariance(6, 6, 1e-09);
```

Comment:
does setting all the data members really matter? IIUC the logic we never actually return a fake beamspot, so the only important point is to set the type, no?

Reply:
removed